File: MyAnimeListJava/src/main/java/com/example/myanimelistjava/controllers/InicioController.java
package com.example.myanimelistjava.controllers;

import com.example.myanimelistjava.MyAnimeListApplication;
import com.example.myanimelistjava.configurations.ViewConfig;
import javafx.application.Platform;
import javafx.event.ActionEvent;
import javafx.fxml.FXML;
import javafx.fxml.FXMLLoader;
import javafx.scene.Scene;
import javafx.scene.control.Hyperlink;
import javafx.scene.control.PasswordField;
import javafx.scene.control.TextField;
import javafx.stage.Stage;

import java.io.IOException;

public class InicioController {
    @FXML
    TextField txtUsername;
    @FXML
    PasswordField txtPassword;
    @FXML
    TextField txtUsernameRegister;
    @FXML
    TextField txtEmail;
    @FXML
    PasswordField txtPasswordRegister;
    @FXML
    PasswordField txtConfirmPassword;
    @FXML
    Hyperlink btnRegister;
    public void changeSceneRegister(ActionEvent actionEvent) throws IOException {
        Platform.setImplicitExit(true);
        var stage = (Stage) btnRegister.getScene().getWindow();
        FXMLLoader fxmlLoader= new FXMLLoader(MyAnimeListApplication.class.getResource("views/registro-view.fxml"));
        var scene = new Scene(fxmlLoader.load(), Double.parseDouble(ViewConfig.WIDTH.getValue()), Double.parseDouble(ViewConfig.HEIGHT.getValue()));
        stage.setTitle("Registro");
        stage.setResizable(false);
        stage.setScene(scene);
        stage.show();
    }
}

File: MyAnimeList/src/main/kotlin/com/example/myanimelist/controllers/InicioController.kt
package com.example.myanimelist.controllers

import com.example.myanimelist.MyAnimeListApplication
import com.example.myanimelist.utils.ViewConfig
import javafx.application.Platform
import javafx.event.ActionEvent
import javafx.fxml.FXML
import javafx.fxml.FXMLLoader
import javafx.scene.Scene
import javafx.scene.control.Hyperlink
import javafx.scene.control.PasswordField
import javafx.scene.control.TextField
import javafx.stage.Stage

class InicioController {
    @FXML
    lateinit var txtUsername: TextField
    @FXML
    lateinit var txtPassword: PasswordField
    @FXML
    lateinit var txtUsernameRegister: TextField
    @FXML
    lateinit var txtEmail: TextField
    @FXML
    lateinit var txtPasswordRegister: PasswordField
    @FXML
    lateinit var txtConfirmPassword: PasswordField
    @FXML
    lateinit var btnRegister: Hyperlink


    fun changeSceneRegister(actionEvent: ActionEvent) {
        Platform.setImplicitExit(true)
        val stage = btnRegister.scene.window as Stage
        val fxmlLoader = FXMLLoader(MyAnimeListApplication::class.java.getResource("views/registro-view.fxml"))
        val scene = Scene(fxmlLoader.load(), ViewConfig.WIDTH.value.toDouble(), ViewConfig.HEIGHT.value.toDouble())
        stage.title = "Registro"
        stage.isResizable = false
        stage.scene = scene
        stage.show()
    }
}